lib/ansible/modules/network/cloudengine/ce_ntp.py

```diff
@@ -0,0 +1,621 @@
+
+    def get_end_state(self):
+        """Get end state info"""
+
+        if self.address:
+            self.end_state = self.get_ntp_exist_config()
+
+    def get_update_cmd(self):
+        """Get updated commands"""
+
+        if self.conf_exsit:
+            return
+
+        cli_str = ""
+        if self.state == "present":
+            if self.address:
+                if self.peer_type == 'Server':
+                    if self.ip_ver == "IPv4":
+                        cli_str = "%s %s" % (
+                            "ntp unicast-server", self.address)
+                    else:
+                        cli_str = "%s %s" % (
+                            "ntp unicast-server ipv6", self.address)
+                elif self.peer_type == 'Peer':
+                    if self.ip_ver == "IPv4":
+                        cli_str = "%s %s" % ("ntp unicast-peer", self.address)
+                    else:
+                        cli_str = "%s %s" % (
+                            "ntp unicast-peer ipv6", self.address)
+
+                if self.key_id:
+                    cli_str = "%s %s %s" % (
+                        cli_str, "authentication-keyid", self.key_id)
+                if self.interface:
+                    cli_str = "%s %s %s" % (
+                        cli_str, "source-interface", self.interface)
+                if (self.vpn_name) and (self.vpn_name != '_public_'):
+                    cli_str = "%s %s %s" % (
+                        cli_str, "vpn-instance", self.vpn_name)
+                if self.is_preferred == "enable":
+                    cli_str = "%s %s" % (cli_str, "preferred")
+        else:
+            if self.address:
+                if self.peer_type == 'Server':
+                    if self.ip_ver == "IPv4":
+                        cli_str = "%s %s" % (
+                            "undo ntp unicast-server", self.address)
+                    else:
+                        cli_str = "%s %s" % (
+                            "undo ntp unicast-server ipv6", self.address)
+                elif self.peer_type == 'Peer':
+                    if self.ip_ver == "IPv4":
+                        cli_str = "%s %s" % (
+                            "undo ntp unicast-peer", self.address)
+                    else:
+                        cli_str = "%s %s" % (
+                            "undo ntp unicast-peer ipv6", self.address)
+                if (self.vpn_name) and (self.vpn_name != '_public_'):
+                    cli_str = "%s %s" % (cli_str, self.vpn_name)
+
+        self.updates_cmd.append(cli_str)
+
+    def work(self):
+        """Excute task"""
+
+        self.get_existing()
+        self.get_proposed()
+
+        self.config_ntp()
+
+        self.get_update_cmd()
+        self.get_end_state()
+        self.show_result()
+
+
+def main():
+    """Main function entry"""
+
+    argument_spec = dict(
+        server=dict(type='str'),
+        peer=dict(type='str'),
+        key_id=dict(type='str'),
```

Comment:
Because the device side configuration is all type ```string```, avoid repeated conversion, Let's keep it.